### PR TITLE
Fix/replay video not found

### PR DESCRIPTION
**Pull request by Animesh404**
**What kind of change does this PR introduce?**

a bugfix for #824 

**Summary**

Earlier while deleting, `delete_recording` was just removing `Recording`. Now, it removes `ActionEvent` as well as `Screenshot`. This ensures all related data is cleaned up correctly.

**Checklist**
* [x] My code follows the style guidelines of OpenAdapt
* [x] I have performed a self-review of my code
* [x] If applicable, I have added tests to prove my fix is functional/effective
* [x] I have linted my code locally prior to submission
* [x] I have commented my code, particularly in hard-to-understand areas
* [ ] I have made corresponding changes to the documentation (e.g. README.md, requirements.txt)
* [ ] New and existing unit tests pass locally with my changes

**How can your code be run and tested?**
1. Create a recording with some description (eg: "testing calculator")
2. Delete this recording and create a new recording with same description ("testing calculator")
3. Run the command given below.

```bash
python -m openadapt.replay VisualReplayStrategy --instructions "calculate 7+8-2"
```

### CODE REVIEW

File: openadapt/db/crud.py [hunk 1]
```diff
@@ -271,6 +271,11 @@ def delete_recording(session: SaSession, recording: Recording) -> None:
         recording (Recording): The recording object.
     """
     recording_timestamp = recording.timestamp
+
+    # Delete associated ActionEvent and Screenshot entries
+    session.query(ActionEvent).filter(ActionEvent.recording_id == recording.id).delete()
+    session.query(Screenshot).filter(Screenshot.recording_id == recording.id).delete()
+
```
Comment (abrichr) [inline openadapt/db/crud.py:278]: What do you think about this intead:

```
class Recording(Base):
    __tablename__ = 'recording'
    id = Column(Integer, primary_key=True)
    timestamp = Column(DateTime)
    action_events = relationship("ActionEvent", cascade="all, delete-orphan", backref="recording")
    screenshots = relationship("Screenshot", cascade="all, delete-orphan", backref="recording")
```

**Review by abrichr — COMMENTED** [review]
Thank you @Animesh404 !

File: openadapt/db/crud.py (hunk 1 above)
Comment (Animesh404) [reply openadapt/db/crud.py]: Done here 09314d7bf478855d06bc265c3c0c9459b8591331

**Review by abrichr — APPROVED** [review]
LGTM!

**Pull request merged**